src/i2c.rs

```diff
@@ -19,16 +19,38 @@ crate::embedded_hal_error!(
     embedded_hal::i2c::ErrorKind
 );
 
+#[cfg(any(esp32, esp32s2))]
 const APB_TICK_PERIOD_NS: u32 = 1_000_000_000 / 80_000_000;
+#[cfg(not(any(esp32, esp32s2)))]
+const XTAL_TICK_PERIOD_NS: u32 = 1_000_000_000 / XTAL_CLK_FREQ;
```

Comment:
Here I'm not sure, but you should check.

Reply:
its 25ns per clock actually